src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersisterProvider.cs

```diff
@@ -38,14 +39,9 @@ internal sealed class LanguageSettingsPersisterProvider : IOptionPersisterProvid
         public async ValueTask<IOptionPersister> GetOrCreatePersisterAsync(CancellationToken cancellationToken)
         {
             if (_lazyPersister is not null)
-            {
                 return _lazyPersister;
-            }
 
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-            var textManager = (IVsTextManager4?)await _serviceProvider.GetServiceAsync(typeof(SVsTextManager)).ConfigureAwait(true);
-            Assumes.Present(textManager);
+            var textManager = await _serviceProvider.GetServiceAsync<SVsTextManager, IVsTextManager4>().ConfigureAwait(false);
 
             _lazyPersister ??= new LanguageSettingsPersister(_threadingContext, textManager, _optionService);
```

Comment:
This line ran on the UI thread before, and now it may be on a background thread. Is that ok?

Reply:
If not, I think you'll have other places to fix in this PR too.

Reply:
i thought i reviewed these for safety.  but it turns out this is not safe.  LanguageSettingsPersister's constructor does need to run on the UI thread.  le-sigh.

Reply:
nope.  this case is very very broken.  it hought i checked all the code, but i missed a case here.